build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,375 @@
+                                pubAccessible: true,
+                                doNotWaitForPreviousBuild: true,
+                                uploadArtifactsOnlyIfSuccessful: true)
+    } catch (error) {
+        echo "ERROR : publishPackages: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+timestamps {
+    node("${env.BotLabel}") {
+        def scmVars
+
+        stageStatus = 0
+        try {
+            timeout(time: 60, unit: 'MINUTES') {    // Time ranges from seconds to minutes depending on how many changes need to be brought down
+                stage('checkout') {
+                    dir(XADir) {
+                        scmVars = checkout scm
+                    }
+
+                    echo "Stage result: checkout: ${currentBuild.currentResult}"
+                }
+            }
+        } catch (error) {
+            echo "ERROR : checkout: Unexpected error: ${error}"
+            currentBuild.result = 'FAILURE'
+        }
+
+        stageStatus = 0
+        try {
+            timeout(time: 30, unit: 'SECONDS') {    // Typically takes less than a second
+                stage('init') {
+                    // Note: PR plugin environment variable settings available here: https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin
+                    isPr = env.ghprbActualCommit != null
+                    def branch = isPr ? env.GIT_BRANCH : scmVars.GIT_BRANCH
+                    def commit = isPr ? env.ghprbActualCommit : scmVars.GIT_COMMIT
+
+                    def buildType = isPr ? 'PR' : 'CI'
+
+                    echo "Job: ${env.JOB_BASE_NAME}"
+                    echo "Branch: ${branch}"
+                    echo "Commit: ${commit}"
+                    echo "Build type: ${buildType}"
+                    if (isPr) {
+                        echo "PR id: ${env.ghprbPullId}"
+                        echo "PR link: ${env.ghprbPullLink}"
+                    }
+
+                    publishBuildFilePaths = "xamarin-android/xamarin.android-oss*.zip,xamarin-android/bin/${env.BuildFlavor}/bundle-*.zip,xamarin-android/bin/Build*/Xamarin.Android.Sdk*.vsix,xamarin-android/prepare-image-dependencies.sh,xamarin-android/build-status*,xamarin-android/xa-build-status*";
+                    publishTestFilePaths = "xamarin-android/xa-test-errors*"
+
+                    echo "Stage result: init: ${currentBuild.currentResult}"
+                }
+            }
+        } catch (error) {
+            echo "ERROR : init: Unexpected error: ${error}"
+            currentBuild.result = 'FAILURE'
+        }
+
+        stageStatus = 0
+        try {
+            timeout(time: 30, unit: 'SECONDS') {    // Typically takes less than a second
+                stage('clean') {
+                    dir('xamarin-android@tmp') {
+                        deleteDir()
+                    }
+
+                    if (fileExists('TestResult.xml')) {
```

Comment:
This seems odd; this is "topdir", right?  How could this file ever exist?

Reply:
Fixed with commit https://github.com/xamarin/xamarin-android/pull/2603/commits/a85b99da53b63966bcbb2f7750757dac77806984
The clean steps were carbon copied from a separate jenkinsfile where the same exact steps don't apply here. A xamarin-android_tmp directory does get generated and so updated that cleaning step to use the correct directory name.